src/components/menu/menu.spec.tsx

```diff
@@ -297,4 +268,32 @@ describe("Menu", () => {
       }
     );
   });
+
+  describe("when a MenuItem unmounts", () => {
+    const MockMenu = () => {
+      const [show, setShow] = React.useState(true);
+
+      return (
+        <Menu>
+          {show && (
+            <MenuItem onClick={() => setShow(false)}>test element 1</MenuItem>
+          )}
+          <MenuItem href="#">test element 2</MenuItem>
+        </Menu>
+      );
+    };
+    it("unregisters itself to the parent Menu", () => {
+      wrapper = mount(<MockMenu />);
+
+      act(() => {
+        wrapper
+          .find(StyledMenuItemWrapper)
+          .first()
+          .props()
+          .onClick({} as React.MouseEvent<HTMLButtonElement>);
```

Comment:
Similar to the previous comments
```suggestion
        wrapper
          .find(StyledMenuItemWrapper)
          .first()
          .simulate("click")
```

Reply:
I don't actually need the type assertion here so I'll just go with removing it for now as this will all be refactored to RTL soon